### PR TITLE
Remove fixed width block

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by findepi** [comment]
> Add Int128ArrayBlock

Does this relate to long `decimal`?

**Comment by dain** [comment]
> > Add Int128ArrayBlock
> 
> Does this relate to long `decimal`?

Yes, it is used for long decimal instead of the FixedWidthBlock

**Comment by sopel39** [comment]
skimmed for now, looks good

**Comment by sopel39** [comment]
Is this preparatory work or just cleanup?

**Comment by dain** [comment]
Both.  It makes it easier to build decimal blocks in some cases that I have been playing with.

File: presto-spi/src/main/java/io/prestosql/spi/block/Int128ArrayBlock.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+        return new Int128ArrayBlock(positionOffset + arrayOffset, length, valueIsNull, values);
+    }
+
+    @Override
+    public Block copyRegion(int positionOffset, int length)
+    {
+        checkValidRegion(getPositionCount(), positionOffset, length);
+
+        positionOffset += arrayOffset;
```
Comment (sopel39) [inline presto-spi/src/main/java/io/prestosql/spi/block/Int128ArrayBlock.java]: you probably shouldn't multiply `arrayOffset` * 2, below you have code:
```
positionOffset * 2
```
which resolves to: `(positionOffset + arrayOffset) * 2`
but I think it should be
```
arrayOffset + positionOffset * 2
```

File: presto-spi/src/test/java/io/prestosql/spi/predicate/TestEquatableValueSet.java [hunk 2]
```diff
@@ -111,7 +113,9 @@ public void testMultipleValues()
         assertFalse(equatables.isAll());
         assertFalse(equatables.isSingleValue());
         assertTrue(equatables.isWhiteList());
-        assertTrue(Iterables.elementsEqual(equatables.getValues(), ImmutableList.of(1L, 2L, 3L)));
+        Collection<Object> values = equatables.getValues();
+        assertEquals(values.size(), 3);
+        assertTrue(values.containsAll(ImmutableSet.of(1L, 2L, 3L)));
```
Comment (sopel39) [inline presto-spi/src/test/java/io/prestosql/spi/predicate/TestEquatableValueSet.java]: why did the semantic change (from `elementsEqual` to `containsAll`)?

File: presto-main/src/main/java/io/prestosql/operator/EnforceSingleRowOperator.java [hunk 3]
```diff
@@ -63,7 +61,7 @@ public OperatorFactory duplicate()
         }
     }
 
-    private static final Page SINGLE_NULL_VALUE_PAGE = new Page(1, new FixedWidthBlock(0, 1, EMPTY_SLICE, Optional.of(Slices.wrappedBooleanArray(true))));
+    private static final Page SINGLE_NULL_VALUE_PAGE = new Page(1, new ByteArrayBlock(1, Optional.of(new boolean[] {true}), new byte[1]));
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/EnforceSingleRowOperator.java]: unrelated, but there seem to assumption that `EnforceSingleRowOperator` produces only single column (single scalar). Is it always the case @kokosing ?

Comment (kokosing) [reply presto-main/src/main/java/io/prestosql/operator/EnforceSingleRowOperator.java]: Yes. Nothing changed here.

File: presto-main/src/main/java/io/prestosql/type/IpAddressType.java [hunk 4]
```diff
@@ -51,19 +93,26 @@ public boolean isOrderable()
...
+        }
+        // compare low order bits
+        return Long.compare(leftBlock.getLong(leftPosition, 0), rightBlock.getLong(rightPosition, 0));
     }
 
     @Override
     public long hash(Block block, int position)
     {
-        return block.hash(position, 0, getFixedSize());
+        return XxHash64.hash(block.getLong(position, 0)) ^ XxHash64.hash(block.getLong(position, SIZE_OF_LONG));
```
Comment (dain) [inline presto-main/src/main/java/io/prestosql/type/IpAddressType.java]: This must match `HASH` operator implementation.

File: presto-main/src/main/java/io/prestosql/type/UnknownType.java [hunk 5]
```diff
@@ -32,7 +37,40 @@ private UnknownType()
...
+    public int getFixedSize()
+    {
+        return Byte.BYTES;
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedBytesPerEntry)
+    {
+        int maxBlockSizeInBytes;
+        if (blockBuilderStatus == null) {
```
[the remark this answers is not in the document]
Comment (dain) [reply presto-main/src/main/java/io/prestosql/type/UnknownType.java]: I think this is in every builder, so maybe it should be.

File: presto-spi/src/test/java/io/prestosql/spi/predicate/TestEquatableValueSet.java (hunk 2 above)
Comment (dain) [reply presto-spi/src/test/java/io/prestosql/spi/predicate/TestEquatableValueSet.java]: When I wrote this code there was no ordering guarantee in `EquatableValueSet`.  I see that it is changed now, so I will revert this part.

**Pull request merged**